src_ops_metrics/pre_processing.py

```diff
@@ -0,0 +1,279 @@
+
+            # Consider all approved reviews
+            pr_approved_dt = [
+                datetime.fromtimestamp(review["submitted_at"])
+                for review in pr["reviews"].values()
+                if review["state"] == "APPROVED"
+            ]
+
+            if pr_approved_dt:
+                # Take maximum to consider last approved if more than one contributor has to approve
+                dt_approved = max(pr_approved_dt)
+
+                ttr_per_pr.append((dt_approved - dt_created).total_seconds() / 3600)
+
+                ttr_in_time.append((dt_created, pr_id, (dt_approved - dt_created).total_seconds() / 3600, pr["size"]))
+                mttr_in_time.append((dt_created, pr_id, np.median(ttr_per_pr)))
+
+            else:
+                dt_merged = datetime.fromtimestamp(pr["merged_at"])
+                ttr_per_pr.append((dt_merged - dt_created).total_seconds() / 3600)
+
+                ttr_in_time.append((dt_created, pr_id, (dt_merged - dt_created).total_seconds() / 3600, pr["size"]))
+                mttr_in_time.append((dt_created, pr_id, np.median(ttr_per_pr)))
+
+            tfr_in_time.append((dt_created, pr_id, (dt_first_review - dt_created).total_seconds() / 3600, pr["size"]))
+
+            mtfr_in_time.append((dt_created, pr_id, np.median(tfr_per_pr)))
+
+            project_prs_size_encoded.append(convert_score2num(label=pr["size"]))
+
+            time_reviews += dt_all_reviews
+
+        if pr["created_by"] not in contributors:
+            contributors.append(pr["created_by"])
+
+    project_reviews_data = {}
+    project_reviews_data["TFR_in_time"] = tfr_in_time
+    project_reviews_data["TTR_in_time"] = ttr_in_time
+    project_reviews_data["MTFR_in_time"] = mtfr_in_time
+    project_reviews_data["MTTR_in_time"] = mttr_in_time
+    project_reviews_data["contributors"] = contributors
+    project_reviews_data["last_review_time"] = max(time_reviews)
+
+    # Encode Pull Request sizes for the contributor
+    project_pr_median_size, project_length_score = convert_num2label(score=np.median(project_prs_size_encoded))
+    project_reviews_data["median_pr_length"] = project_pr_median_size
+    project_reviews_data["median_pr_length_score"] = project_length_score
+
+    return project_reviews_data
+
+
+def pre_process_contributors_data(data: Dict[str, Any]):
+    """Pre process of data for contributors in a project repository."""
+    pr_ids = sorted([int(k) for k in data.keys()])
+
+    contributors_reviews_data = {}
+
+    tfr_per_pr = {}  # Time to First Review (TTFR) [hr]
+    ttr_per_pr = {}  # Time to Review (TTR) [hr]
+
+    pr_length = {}  # Pull Request length per Reviewer
+
+    mtfr_in_time = {}  # Median TTFR [hr]
+    mttr_in_time = {}  # Mean TTR [hr]
+
+    tfr_in_time = {}  # TTFR in time [hr]
+    ttr_in_time = {}  # TTR in time [hr]
+
+    for pr_id in pr_ids:
+        pr = data[str(pr_id)]
+        if pr["reviews"]:
```

Comment:
I think that a guard should be used in this case to avoid making a big part of the code being wrapped in one if condition